pkg/segment/structs/segsearchstructs.go

```diff
@@ -375,6 +375,26 @@ func extractSearchQueryFromMatchFilter(match *MatchFilter) *SearchQuery {
 	} else {
 		currQuery.SearchType = MatchWords
 	}
+	if len(match.MatchWords) > 0 {
+		for _, word := range match.MatchWords {
+			cval := dtu.ReplaceWildcardStarWithRegex(string(word))
+			rexpC, err := regexp.Compile(cval)
+			if err != nil {
+				log.Errorf("extractSearchQueryFromMatchFilter: regexp compile failed for MatchWord %s, err=%v", word, err)
+			} else {
+				currQuery.MatchFilter.Regexp = rexpC
+			}
+		}
+	} else if match.MatchPhrase != nil {
```

Comment:
this cannot be an else, when matchPhrase is present the parser fills in both matchwords and matchphrase.